service/src/db/dbOperations.ts

```diff
@@ -27,24 +27,40 @@ export async function findArtifactByUrlAndVersion<T extends FhirArtifact>(
 }
 
 /**
- * searches the database and returns an array of all resources of the given type that match the given query
+ * searches the database and returns an array whose object contains metadata and data: an array of resources of the given type that match the given query
  */
-export async function findResourcesWithQuery<T extends FhirArtifact>(
+export async function findResourcesWithQuery<T extends FhirArtifact[]>(
   query: Filter<any>,
   resourceType: ArtifactResourceType
 ) {
   const collection = Connection.db.collection(resourceType);
+  const projection = { _id: 0, _dataRequirements: 0 };
+  const pagination: any = 'skip' in query ? [{ $skip: query.skip }, { $limit: query.limit }] : [];
   query._dataRequirements = { $exists: false };
   query._summary = { $exists: false };
   query._elements = { $exists: false };
-  return collection.find<T>(query, { projection: { _id: 0, _dataRequirements: 0 } }).toArray();
+  query._count = { $exists: false };
+  query.limit = { $exists: false };
+  query.skip = { $exists: false };
+  return collection
```

Comment:
The BSONObj size issue has been addressed! Great idea from @hossenlopp to grab ids only in the facet aggregation, and then to use those ids to pull all of the records needed for whatever page you're on. The record pull then separates the results into their own documents (instead of one big document because of the facet).